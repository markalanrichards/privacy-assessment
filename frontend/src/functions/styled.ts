import * as styledComponents from "styled-components";

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider
} = styledComponents as styledComponents.ThemedStyledComponentsModule<
  IThemeInterface
>;

interface Color {
  primaryColor: string;
}

interface Size {
  large: string;
}

export interface IThemeInterface {
  color: Color;
  size: Size;
}

export const defaultTheme: IThemeInterface = {
  color: {
    primaryColor: "yellow"
  },
  size: {
    large: "10"
  }
};

export default styled;
export { css, injectGlobal, keyframes, ThemeProvider };
