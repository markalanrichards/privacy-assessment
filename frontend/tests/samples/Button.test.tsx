import * as React from "react";
import { shallow, mount } from "enzyme";
import styled, { ThemeProvider } from "styled-components";
import "jest-styled-components";

describe("Unthemed components", () => {
  const Button = styled.button`
    color: red;
  `;

  test("shallow - it works", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  test("wrapper - it works", () => {
    const wrapper = mount(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
});

test("With Theme", () => {
  const defaultTheme = {
    main: "mediumseagreen"
  };

  const Button = styled.button`
    color: ${props => props.theme.main};
  `;
  const shallowWithTheme = (tree, theme) => {
    const context = shallow(<ThemeProvider theme={theme} />)
      .instance()
      // @ts-ignore
      .getChildContext(); // not in enzyme types file GRRRR!!
    return shallow(tree, { context });
  };

  const wrapper = shallowWithTheme(<Button />, defaultTheme);
  // throw wrapper.debug()
  expect(wrapper).toMatchSnapshot();
});
