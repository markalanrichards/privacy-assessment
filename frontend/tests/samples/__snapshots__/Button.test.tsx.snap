// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unthemed components shallow - it works 1`] = `
.c0 {
  color: red;
}

<button
  className="c0"
/>
`;

exports[`Unthemed components wrapper - it works 1`] = `
.c0 {
  color: red;
}

<styled.button>
  <button
    className="c0"
  />
</styled.button>
`;

exports[`With Theme 1`] = `
.c0 {
  color: mediumseagreen;
}

<button
  className="c0"
/>
`;
